### PR TITLE
Web/UI/AdvancedUI Recorders

### PR DESCRIPTION
**Pull request by openbots-ff**
Web Recorder

- Produces a sequence of clicks/types/navigations executed using the selenium web commands.
- Element values are now saved as a DataTable consisting of enabled status, parameter name, and parameter values

UI Recorder

- Produces a sequence of clicks/types/window activations using mouse coordinates and basic input commands.

Advanced UI Recorder

- Produces a sequence of clicks/types executed using the UI Automation command which utilizes windows automation to find selectors in wpf/winform applications. 

TODO 
- Utilize surface automation in the Advanced UI Recorder wherever selectors are not found, as well as provide the option to solely use surface automation.
- Add support for hotkeys in all the above recorders
- Add the ability to locate windows and save as an activate window command in the advanced UI recorder.

### CODE REVIEW

File: taskt.Engine/AutomationEngineInstance.cs [hunk 1]
```diff
@@ -214,7 +214,7 @@ private void ExecuteScript(string data, bool dataIsFile)
 
                         if (elementFound != null)
                         {
-                            elementFound.ElementType = elem.ElementType;
+                            //elementFound.ElementType = elem.ElementType;
```
Comment (openbots-ab) [inline taskt.Engine/AutomationEngineInstance.cs:217]: Please remove the comment.

File: taskt.Studio/Commands/Web Browser Commands/SeleniumElementActionCommand.cs [hunk 2]
```diff
@@ -76,9 +56,18 @@ public class SeleniumElementActionCommand : ScriptCommand
                      "\n\tCSS Selector: [attribute=value]" +
                      "\n\tLink Text: https://www.mylink.com/"
                     )]
+        [Remarks("If multiple parameters are enabled, an attempt will be made to find an element(s) that matches all the selected parameters.")]
```
Comment (openbots-ab) [inline taskt.Studio/Commands/Web Browser Commands/SeleniumElementActionCommand.cs:59]:
```suggestion
        [Remarks("If multiple parameters are enabled, an attempt will be made to find an element that matches all the selected parameters.")]
```

**Review by openbots-ab — CHANGES_REQUESTED** [review]
Please make a couple of minor changes and it looks good to me.

File: taskt.Engine/AutomationEngineInstance.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt.Engine/AutomationEngineInstance.cs]: done

File: taskt.Studio/Commands/Web Browser Commands/SeleniumElementActionCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt.Studio/Commands/Web Browser Commands/SeleniumElementActionCommand.cs:59]: no, element(s) is right. You have the option of "find element" or "find elements" which returns a list of matching elements.

**Review by openbots-ab — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**